### PR TITLE
Build: Use new Travis-CI Docker infrastrucutre

### PR DESCRIPTION
**Pull request by nschonni**
Moves the build worker to the new container based version http://docs.travis-ci.com/user/migrating-from-legacy/

Notes;
- Not ensure that the Bundler caching is working full, but it did seem to speed up my build by 30% after the first build primed the asset cache.
- Since we run `npm install` rather than `npm update` in the scripts, this will keep the versions at a set install version while the semver is still respected
- The global npm installs (grunt-cli, bower) don't appear to be cached right now, but since the location is dynamic based on the version of Node that nvm loads, I parked trying to address that.

### CODE REVIEW

**Comment by nschonni** [comment]
First run here was " ran for 14 min 56 sec"
Reset just to sample the difference

File: .travis.yml [hunk 1]
```diff
@@ -20,3 +22,10 @@ after_success:
 notifications:
   irc:
   - "chat.freenode.net#wet-boew"
+
+cache:
+  bundler: true
+  directories:
+  - node_modules
+  - lib
+  - vendor/cache
```
Comment (LaurentGoderre) [inline .travis.yml]: Does this folder exists anywhere? Also how does it know how to invalidate the cache?

**Comment by nschonni** [comment]
After restarting "ran for 9 min 39 sec", so ~30% faster on subsequent builds to a branch. This may not speed up single build PRs though

File: .travis.yml (hunk 1 above)
Comment (nschonni) [inline .travis.yml]: You can see it here https://travis-ci.org/wet-boew/wet-boew/caches, or by selecting "Caches" under the Settings dropdown

**Comment by LaurentGoderre** [comment]
The benefit of not caching is the build would catch problems introduce by children dependency updated., basically it's acting like a new developer setup would.

**Comment by nschonni** [comment]
> The benefit of not caching is the build would catch problems introduce by children dependency updated

Yeah, I mentioned that in the description.

>> Since we run npm install rather than npm update in the scripts, this will keep the versions at a set install version while the semver is still respected

Changing to `npm update` would bump the out of date installs, but not the dependencies that are already ok.

**Comment by LaurentGoderre** [comment]
But not the nested dependencies, which is usually where problems come from. I guess it is a minor issue because we still wipe our node_modules in the rare cases where this happens

**Pull request merged**